File: tests/rust/testA.rs
/*
    This is a test file for testing the rust code.
    This file contains two functions that are identical except for the comments.
    The test will check if the two functions are identical.
*/
fn print_hello_test_a1() {
    println!("Hello, World!");
    for i in 0..5 {
        println!("This is line {}", i);
        if i % 2 == 0 {
            println!("Even number");
        } else {
            // This is an odd number
            println!("Odd number");
        }
    }
}

fn print_hello_test_a2() {
    println!("Hello, World!");
    // for loop
    for i in 0..5 {
        // print line number
        println!("This is line {}", i); // print line number
        if i % 2 == 0 {
            // even number
            println!("Even number");
        } else {
            // odd number
            println!("Odd number");
        }
    }
}

fn main() {
    print_hello_test_a1();
    print_hello_test_a2();
} 

File: tests/java/TestA.java
public class TestA {
    // This method prints "Hello, World!" and then prints lines with even/odd checks
    public static void printHelloTestA1() {
        System.out.println("Hello, World!"); // Print greeting
        for (int i = 0; i < 5; i++) { // Loop from 0 to 4
            System.out.println("This is line " + i); // Print line number
            if (i % 2 == 0) { // Check if the number is even
                System.out.println("Even number"); // Print if even
            } else {
                System.out.println("Odd number"); // Print if odd
            }
        }
    }

    // This method is similar to printHelloTestA1
    public static void printHelloTestA2() {
        System.out.println("Hello, World!"); // Print greeting
        for (int i = 0; i < 5; i++) { // Loop from 0 to 4
            System.out.println("This is line " + i); // Print line number
            if (i % 2 == 0) { // Check if the number is even
                System.out.println("Even number"); // Print if even
            } else {
                System.out.println("Odd number"); // Print if odd
            }
        }
    }

    /*
     * Main method to execute the test methods
     */
    public static void main(String[] args) {
        printHello(); // Call to a method not defined in this snippet
        printHelloTestA1(); // Call the first test method
        printHelloTestA2(); // Call the second test method
    }
}